cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
+	end = regexp.MustCompile(`(?i)\s+(-?(?:filetype|package|pkg|path|file)):(\S+)\s*$`)
+)
+
+func parseFilters(query *url.Values, qstr string, atstart bool) string {
+	q := *query
+
+	var re *regexp.Regexp
+	if atstart {
+		re = start
+	} else {
+		re = end
+	}
+
+	matches := re.FindStringSubmatchIndex(qstr)
+	for matches != nil {
+		// matches is [token_start, token_end, type_start, type_end, value_start, value_end]
```

Comment:
I don’t quite follow. Both of these functions give you the submatches, it’s just that one of them returns the index, and the other one returns strings that correspond to the ones you get by slicing. Am I misunderstanding your code? Where does the need to use `strings.TrimPrefix` arise?